TestFoundation/TestNSHTTPCookieStorage.swift

```diff
@@ -227,18 +227,22 @@ class TestNSHTTPCookieStorage: XCTestCase {
         storage.setCookie(testCookie)
         XCTAssertEqual(storage.cookies!.count, 3)
         var destPath: String
+        //let bundleName = Bundle.main.object(forInfoDictionaryKey: "CFBundleDisplayName") as! String
+        //TODO: Above api returns nil on Linux when this code is introduced. Hence, using the below workaround.
+        //Needs replacement once fixed.
+        let bundlePath = Bundle.main.bundlePath
+        var bundlePathComponents = bundlePath.components(separatedBy: "/")
+        let bundleName: String = "/" + bundlePathComponents[bundlePathComponents.count - 1]
         if let xdg_data_home = getenv("XDG_DATA_HOME") {
-            destPath = String(utf8String: xdg_data_home)! + "/.cookies.shared"
+            destPath = String(utf8String: xdg_data_home)! + bundleName + "/.cookies.shared"
         } else {
-            destPath = NSHomeDirectory() + "/.local/share/.cookies.shared"
+            destPath = NSHomeDirectory() + "/.local/share" + bundleName + "/.cookies.shared"
```

Comment:
"/" is appended to the bundleName during initialisation

`var bundleName = "/" + bundlePath.components(separatedBy: "/").last!`